spec/system/country_page_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.describe "Country page", type: :system, driver: :selenium_chrome, js: true do
+  include_context 'countries_setup'
+
+  let(:country_path) { country_path = GeoEntity.last.name.downcase.gsub(/ /, '-') }
+
+  before do
+    allow(GeoEntity).to receive(:permitted_countries).and_return(GeoEntity.all)
+    country
+    many_species
+  
+    visit "/#{country_path}"
+  end
+
+  scenario 'Visiting a country page' do
+    expect(page).to have_content(GeoEntity.last.actual_name)
+  end
+
+  scenario 'Finding the total length of the coastline' do
+    chart_title = find('.habitat-representation__chart-title')
+
+    unrounded_coastline = GeoEntity.last.coastal_stat.total_coast_length
+    coastline_length = ActiveSupport::NumberHelper.number_to_delimited(unrounded_coastline.round)
+
+    expect(chart_title.text).to eq("Percentage of coastline (#{coastline_length} km) covered by habitats")
+  end
+
+  scenario 'Looking at example of species section' do
+    expect(page).to have_content('Total number of species: 20')
+
+    species_common_names = GeoEntity.last.species.take(3).pluck(:common_name)
+
+    expect(page).to have_content(species_common_names[0])
+    expect(page).to have_content(species_common_names[1])
+    expect(page).to have_content(species_common_names[2])
+  end
+
+  scenario 'Opening the main national data sources modal' do
+    find('.downloads__li > #modal-trigger-citation').click
+
+    expect(page).to have_css('.modal__dialog')
```

Comment:
Is there a way to easily check that the modal__dialog is visible? E.g. if it were changed to use v-show for some reason, would this then always pass? I'd be tempted to check that it isn't visible before and then is visible afterwards, to be sure that a pass corresponds to it working properly.

Also, is it worth checking for the 'Sources'  title (or even better one of the sources), to check that it isn't some other modal that has accidentally been opened (which could happen if the js was broken as the modals are shared).